Core/LLVM/llvm-3.4/include/llvm/IR/IntrinsicsLunarGLASSTop.td

```diff
@@ -148,6 +148,7 @@ let TargetPrefix = "gla" in {
   def int_gla_smulExtended : Intrinsic<[llvm_anyint_ty, llvm_anyint_ty], [llvm_anyint_ty, llvm_anyint_ty], [IntrNoMem]>;
   def int_gla_uFma : Intrinsic<[llvm_anyint_ty], [llvm_anyint_ty, llvm_anyint_ty, llvm_anyint_ty], [IntrNoMem]>;
   def int_gla_sFma : Intrinsic<[llvm_anyint_ty], [llvm_anyint_ty, llvm_anyint_ty, llvm_anyint_ty], [IntrNoMem]>;
+  def int_gla_ibMix : Intrinsic<[llvm_anyint_ty], [llvm_anyint_ty, llvm_anyint_ty, llvm_anyint_ty], [IntrNoMem]>;

```

Comment:
Possibly. This is partly from recent learning with SPIR-V, which would get mirrored in a SPIR-V LunarGLASS back end that itself mirrors the GLSL back end:  The versions of mix that do Boolean selection should become select.

Now, the integer versions of mix that are not Boolean driven are the null set:  there is no such thing as an arithmetical mix of integer operands.

Last thing to keep in mind is backward compatibility, if getting to full correctness means removing something that's been there for a while, it should stay anyway, but all the new stuff should make sense according to this thread.